consensus/src/chained_bft/chained_bft_smr.rs

```diff
@@ -214,24 +214,8 @@ impl<T: Payload> StateMachineReplication for ChainedBftSMR<T> {
         let consensus_state = initial_data.state();
         let highest_timeout_certificates = initial_data.highest_timeout_certificates().clone();
         if initial_data.need_sync() {
-            loop {
-                // make sure we sync to the root state in case we're not
-                let status = block_on(state_computer.sync_to(initial_data.root_ledger_info()));
-                match status {
-                    Ok(true) => break,
-                    Ok(false) => panic!(
-                    "state synchronizer failure, this validator will be killed as it can not \
-                 recover from this error.  After the validator is restarted, synchronization will \
-                 be retried.",
-                ),
-                    Err(e) => panic!(
-                    "state synchronizer failure: {:?}, this validator will be killed as it can not \
-                 recover from this error.  After the validator is restarted, synchronization will \
-                 be retried.",
-                    e
-                ),
-                }
-            }
+            // make sure we sync to the root state in case we're not
+            state_computer.sync_to_or_bail(initial_data.root_ledger_info());
```

Comment:
did you remove the loop by accident or intention?

Reply:
Intention: if, as I suspect (and @dmitri-perelman, @aching  seemed to confirm), nothing is executing concurrently of this, there's nothing else to make `need_sync()` compare to a different target and require it.
does that make sense to you?

Reply:
the loop here is not about concurrency. it was introduced as a hack to bypass the contention with network start-up.
if we hit this code before network connect to enough peers, we'll panic and restart and result in a crash-loop.

Reply:
although it seems changed after - it used to have a sleep if the download failed, I guess the logic moved into state synchronizer now so we're probably good.